app/audit/models/ClaimARefundAuditModel.scala

```diff
@@ -43,17 +44,16 @@ case class ClaimARefundAuditModel(balanceDetails: Option[BalanceDetails],
   }
 
   private def getCreditType(credit: (DocumentDetailWithDueDate, FinancialDetail)): String = {
-    val isMFA: Boolean = credit._2.isMFACredit
-    val isCutOverCredit: Boolean = credit._2.isCutOverCredit
+    val creditType: Option[CreditType] = credit._2.getCreditType
     val isPayment: Boolean = credit._1.documentDetail.paymentLot.isDefined
-    val isBalancingCreditCharge: Boolean = credit._2.isBalancingChargeCredit
-
-    (isMFA, isCutOverCredit, isBalancingCreditCharge, isPayment) match {
-      case (true, _, _, _) => "Credit from HMRC adjustment"
-      case (_, true, _, _) => "Credit from an earlier tax year"
-      case (_, _, true, _) => "Balancing charge credit"
-      case (_, _, _, true) => s"Payment made on ${getFullDueDate(credit._1.dueDate.get)}"
-
+    creditType match {
+      case Some(MfaCreditType) => "Credit from HMRC adjustment"
+      case Some(CutOverCreditType) => "Credit from an earlier tax year"
+      case Some(BalancingChargeCreditType) => "Balancing charge credit"
+      case _ if isPayment => s"Payment made on ${getFullDueDate(credit._1.dueDate.get)}"
+      case None =>
+        Logger("application").error(s"[ClaimARefundAuditModel][getCreditType] Missing or non-matching credit: not a valid credit type")
+        "unknownCredit"
       case error =>
```

Comment:
error line become redundant after this change, so you are free to drop it as input would be alway of type => Option[CreditType]